dialogue-core/src/main/java/com/palantir/dialogue/core/PinUntilErrorChannel.java

```diff
@@ -129,11 +129,12 @@ LimitedChannel getCurrentChannel() {
                 .map(future -> DialogueFutures.addDirectCallback(future, new FutureCallback<Response>() {
                     @Override
                     public void onSuccess(Response response) {
-                        if (Responses.isQosStatus(response) || Responses.isServerError(response)) {
+                        // We *only* switch nodes on server *errors*, NOT QoS responses to support transactional
+                        // workflows where it is important for a large number of requests to all land on the same node,
+                        // even if a couple of them get rate limited in the middle.
+                        if (Responses.isServerError(response)) {
                             OptionalInt next = incrementHostIfNecessary(currentIndex);
                             instrumentation.receivedErrorStatus(currentIndex, channel, response, next);
-                            // TODO(dfox): handle 308 See Other somehow, as we currently don't have a host -> channel
```

Comment:
This comment should have been removed, 308 is covered by isQosStatus.I think we want to specifically opt out of the too many requests status here instead of checking isServerError.

Reply:
like an `if (Responses.isQoS(reposnse)) { return; }`?

Reply:
I think we want to add “&& !isTooManyRequests” to the original conditional